### PR TITLE
Conhost: copy RTF to clipboard

### LINKED ISSUE #3560: Connect GenRTF to conhost (Win32Interactivity Clipboard)
Make conhost do RTF!

@DHowett-MSFT @carlos-zamora What needs to be done to bring this into conhost? Are you gonna create a new issue?

_Originally posted by @anirudhrb in https://github.com/microsoft/terminal/pull/3535#issuecomment-553435031_

### PR DESCRIPTION
**Pull request by anirudhrb**
RTF data is now copied to the clipboard. The clipboard format name
used for RTF data is `Rich Text Format`.

Refactored some code in `Clipboard.cpp` so that the code for setting
data to the clipboard is re-used. Also, renamed parameter
`fAlsoCopyHtml` to `fAlsoCopyFormatting` to make it more generic.

Tested by copying text from console to WordPad. Also verified that
HTML copy is not regressed by copying to Word.

Closes #3560.

### CODE REVIEW

**Comment by anirudhrb** [comment]
Please review! @carlos-zamora @DHowett-MSFT

File: src/interactivity/win32/Clipboard.cpp [hunk 1]
```diff
@@ -319,6 +300,35 @@ void Clipboard::CopyTextToSystemClipboard(const TextBuffer::TextAndColor& rows,
     globalHandle.release();
 }
 
+void Clipboard::CopyToSystemClipboard(std::string s, LPCWSTR lpszFormat)
```
Comment (carlos-zamora) [inline src/interactivity/win32/Clipboard.cpp]: Please add comment

Comment (carlos-zamora) [inline src/interactivity/win32/Clipboard.cpp]: Can we rename the parameter to be something more descriptive than `s`? Adding the function description will also help this issue.

Comment (anirudhrb) [reply src/interactivity/win32/Clipboard.cpp]: Done!

Comment (anirudhrb) [reply src/interactivity/win32/Clipboard.cpp]: Done.

File: src/interactivity/win32/Clipboard.cpp [hunk 2]
```diff
@@ -319,6 +300,40 @@ void Clipboard::CopyTextToSystemClipboard(const TextBuffer::TextAndColor& rows,
...
+// - Copies the given string onto the global system clipboard in the sepcified format
+// Arguments:
+// - stringToCopy - The string to copy
+// - lpszFormat - the name of the format
+void Clipboard::CopyToSystemClipboard(std::string stringToCopy, LPCWSTR lpszFormat)
+{
+    const size_t rtfSize = stringToCopy.size() + 1; // +1 for '\0'
+    if (rtfSize)
+    {
+        wil::unique_hglobal globalHandleHTML(GlobalAlloc(GMEM_MOVEABLE | GMEM_DDESHARE, rtfSize));
```
Comment (zadjii-msft) [inline src/interactivity/win32/Clipboard.cpp]: I'd maybe rename this to be more generic
```suggestion
        wil::unique_hglobal globalHandleData(GlobalAlloc(GMEM_MOVEABLE | GMEM_DDESHARE, rtfSize));
```

File: src/interactivity/win32/Clipboard.cpp [hunk 3]
```diff
@@ -319,6 +300,40 @@ void Clipboard::CopyTextToSystemClipboard(const TextBuffer::TextAndColor& rows,
...
 }
 
+// Routine Description:
+// - Copies the given string onto the global system clipboard in the sepcified format
+// Arguments:
+// - stringToCopy - The string to copy
+// - lpszFormat - the name of the format
+void Clipboard::CopyToSystemClipboard(std::string stringToCopy, LPCWSTR lpszFormat)
+{
+    const size_t rtfSize = stringToCopy.size() + 1; // +1 for '\0'
```
Comment (zadjii-msft) [inline src/interactivity/win32/Clipboard.cpp]: I'd probably rename this too
```suggestion
    const size_t cbData = stringToCopy.size() + 1; // +1 for '\0'
```

Just to not mix up "HTML" and "RTF" in the same function

File: src/interactivity/win32/Clipboard.cpp (hunk 2 above)
Comment (anirudhrb) [reply src/interactivity/win32/Clipboard.cpp]: Done

File: src/interactivity/win32/Clipboard.cpp (hunk 3 above)
Comment (anirudhrb) [reply src/interactivity/win32/Clipboard.cpp]: Done

**Pull request merged**